### ISSUE TITLE
Create and Implement New Veteran Status Card Component 

### ISSUE THREAD

**Issue by ajsarkar28** [department-of-veterans-affairs/va-mobile-app#7940]
AC:

**Component Creation**
- A new Veteran Status Card component is created, that reflects the designs in this [figma file](https://www.figma.com/file/q4oTyzyLfYrZVz4CACzVLA/Personalization-Visual-Design---Working---VA-Mobile?type=design&node-id=3046-1978&mode=design&t=vJWw89ED2PwHi60F-4)
- This new component follows the existing function of the Veteran Status Card. This includes click analytics 
- VQA by Lauren and Therese is required to complete this ticket

**Component Implementation**
- The Veteran Status Card component, created above, is implemented on the home screen in the About you section

**Accessibility & Content** 
- All accessibility guidelines in previously linked Figma file have been followed
- All content is linked in this [Mural](https://app.mural.co/t/adhoccorporateworkspace2583/m/adhoccorporateworkspace2583/1704381519703/cd6d78c5824a5fb0cc8b26597f0ad271072df8f2?sender=u4daa07e974be596b6b097258)

**Comment by TKDickson** [comment]
[Test rail cases](https://dsvavsp.testrail.io/index.php?/cases/view/59279)

**Comment by TKDickson** [comment]
Have already fixed a pre-existing bug where the VSC would announce as a link, even to veterans who had a discharge status other than honorable. Which is great. Android even after this fix is still announcing (when you have TalkBack hint text on "double tap to activate", idk if we'll be able to fix that part)

**Comment by TKDickson** [comment]
Have tested Army, Navy, Space Force, and Marine Corps. Still need to test Air Force and Coast Guard (as well as multiple military histories, and no military history of course) I don't expect any issues, but I'd rather check and it's kind of fun.

**Comment by TKDickson** [comment]
All branches (of the military) tested

**Comment by TKDickson** [comment]
Checking with UX (slack thread) about how we can improve text resizing within this component (maybe setting a max size on the seal icon)?

Also - Figma specifies for the VSC this behavior for screenreader:
```
Branch of Service
/ 
Proof of Veteran Status link
```

**Comment by TKDickson** [comment]
I'm just... combining things from this ticket and #7966 into this comment, because they're on the same PR.

**Resolved**
- [FIXED](https://drive.google.com/file/d/1I0c7sxzLGRSLJ9Y3j5p_uCdjVSQgZ7BM/view?usp=sharing): _For veterans with a discharge status other than honorable. Android even after the earlier fix is still announcing (when you have TalkBack hint text on "double tap to activate"); both iOS and Android in voice access show the card as actionable as well. Ideally non-actionable across all assistive tech._
- [FIXED](https://drive.google.com/file/d/1mzvq5wcBq-wwwJ8x5H-8XUxPxxu2t5kB/view?usp=sharing): _Android only, we're seeing the same type of thing in the DR and $ portions as have seen previously (reads as: “disability rating 70% service connected” (swipe) “70%” (swipe) “service connected”)_
- [FIXED](https://drive.google.com/file/d/1I6kPyuXnWm3H1PkHmP2YLe-x6wtquA6x/view?usp=sharing): _Checking with UX (slack thread) about how we can improve text resizing within this component (maybe setting a max size on the seal icon)? UX wants to cap the seal at 40x40_
- New ticket #8143 for this & related items: _Right now, the disability rating and military service history API calls kick off during the sync screen, but the letters beneficiary call only kicks off AFTER the home screen loads. Can we be front-loading the beneficiary call as well, to make loading the $ info on the home screen (or the entire about you section, I think, after #7968 adds the loading stuff) faster?_

**Not yet resolved**
- Binny is looking for UX signoff on spinning up a different ticket to figure out how we're going to handle this: _VSC not matching Figma/design for screen reader behavior:
Branch of Service
/ 
Proof of Veteran Status link_

**Comment by TKDickson** [comment]
Pinged @wavelaurenrussell for VQA on this and same-PR-ticket #7966

**Comment by wavelaurenrussell** [comment]
Discussed 

- Add drop shadow to Vet Status, DR/MP block in light mode
- Vet Status: Check spacing of chevron from the right edge
- Monthly compensation payment x-height seems weird (aka not like the others) ..is there some type of font defect?

**Comment by wavelaurenrussell** [comment]
@dumathane made the changes. Approved.

**Comment by TKDickson** [comment]
WIth QA and VQA signoff, merged to XL. Closing

**Comment by TKDickson** [comment]
[belated test run link](https://dsvavsp.testrail.io/index.php?/runs/view/5289&group_by=cases:section_id&group_order=asc)
